File: lab07/frequency.pl
#!/usr/bin/perl
foreach $file (glob "lyrics/*.txt") { 
	my $total=0;
	my $occur=0;
	open FILE ,'<',$file or die;
	@files=<FILE>;
	close FILE;
	$file =~ s/lyrics\///g;
        $file =~ s/.txt//g;
        $file =~ s/_/ /g;
	foreach $line (@files) {
	        $line = lc $line;
		@all = $line =~ /[A-Za-z]+/g;		
        	@array = $line =~ /\b$ARGV[0]\b/g;   #match a word boundary
        	$total += $#all +1;
		$occur += $#array+1;
	}
	my $result=$occur/$total;
	printf("%4d/%6d = %.9f %s\n","$occur","$total","$result","$file");
}

File: lab07/frequency.py
#!/usr/bin/python
import re, sys, glob
reg= r"\b"+re.escape(sys.argv[1])+r"\b"
for file in sorted(glob.glob("lyrics/*.txt")):
	count=0
	total=0
	content = open(file,"r")
	name = re.sub("lyrics/","",file)
	name = re.sub(".txt","",name)
	name = re.sub("_"," ",name)
	for line in content:
		line = line.lower()
		all = re.findall("[A-Za-z]+",line)
		array = re.findall(reg,line)
		total += len(all)
		count += len(array)
	value = float(count)/float(total)
	print("%4d/%6d = %.9f %s" %(count, total, value, name))
